lib/modules/datasource/bitrise/index.ts

```diff
@@ -0,0 +1,89 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { detectPlatform } from '../../../util/common';
+import { parseGitUrl } from '../../../util/git/url';
+import { GithubHttp } from '../../../util/http/github';
+import { joinUrlParts } from '../../../util/url';
+import { parseSingleYaml } from '../../../util/yaml';
+import { GithubDirectoryResponse } from '../../platform/github/schema';
+import semver from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { BitriseStepFile } from './schema';
+
+export class BitriseDatasource extends Datasource {
+  static readonly id = 'bitrise';
+
+  constructor() {
+    super(BitriseDatasource.id);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/bitrise-io/bitrise-steplib.git',
+  ];
+
+  override readonly releaseTimestampSupport = true;
+  override readonly releaseTimestampNote =
+    'The release timestamp is determined from the `published_at` field in the results.';
+  override readonly sourceUrlSupport = 'release';
+  override readonly sourceUrlNote =
+    'The source URL is determined from the `source_code_url` field of the release object in the results.';
+
+  @cache({
+    namespace: `datasource-${BitriseDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      `${registryUrl}/${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    const client = new GithubHttp(this.id);
+
+    const parsedUrl = parseGitUrl(registryUrl);
+    if (detectPlatform(registryUrl) !== 'github') {
+      logger.warn(
+        `${parsedUrl.source} is not a supported Git hoster for this datasource`,
```

Comment:
I think we should validate GitHub here, as the code below only works for GitHub not for ghes